internal/formatters/results/timeline/formatter.go

```diff
@@ -3,57 +3,126 @@
 package timeline
 
 import (
+	"encoding/json"
 	"fmt"
 	"sort"
-	"strings"
 
 	"github.com/cobaltspeech/sdk-cubic/grpc/go-cubic/cubicpb"
+	"github.com/golang/protobuf/ptypes/duration"
 )
 
-// Format generates a list of formatted utterances.
-// The format of each utterance is "start_time_ms|audio_channel_id|1-best transcript".
-//
-// Some assumptions:
-// * The config used in the original transcribe request included, so that timestamps will be available:
-//   &cubicpb.RecognitionConfig{
-//     EnableWordTimeOffsets: true,
-//     EnableRawTranscript: true
-//   }
+// Alternative is a subset of the fields in cubicpb.RecognitionAlternative
+type Alternative struct {
+	StartTime  int64               `json:"start_time"`
+	Confidence float64             `json:"confidence"`
+	Transcript string              `json:"transcript"`
+	Words      []*cubicpb.WordInfo `json:"-"`
+}
+
+// MarshalJSON hides the unfriendly JSON output for duration.Duration
+func (a *Alternative) MarshalJSON() ([]byte, error) {
+	// create aliases so we have all the same fields but none of the methods and don't inherit the original type's MarshalJSON
```

Comment:
I went through all kinds of contortions to get duration.Duration to serialize nicely, to no avail.  Rather than have a Word type in this package and having yet another nested loop on line 119, I followed the pattern from this blog post: http://choly.ca/post/go-json-marshalling/. It looks crazy, but at least most of the crazy is contained to this function!  If you know a better way of doing it, I'd be delighted!

Reply:
I'm going to skip it for now, and maybe address it in 6 months when I've forgotten why we are doing it this way.